design/src/rfcs/rfc0031_providing_fallback_credentials_on_timeout.md

```diff
@@ -0,0 +1,251 @@
+RFC: Providing fallback credentials on external timeout
+=======================================================
+
+> Status: RFC
+>
+> Applies to: client
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+This RFC proposes a fallback mechanism for credentials providers on external timeout (see the [Terminology](#terminology) section), allowing them to continue serving (possibly stale) credentials for the sake of overall reliability of the intended service; the IMDS credentials provider is an example that must fulfill such a requirement to support static stability.
+
+Terminology
+-----------
+
+- External timeout: A timeout within which a call to the `provide_credentials` method must complete. `provide_credentials` returns a future and if it does not complete within this timeout, it gets dropped on the floor and there is no output retrieved from the future.
```

Comment:
This sentence felt a bit wordy to me so I tried my hand at re-wording it. Let me know if you think it makes anything clearer.

```suggestion
- External timeout: The name of the timeout that occurs when a duration elapses before an async call to `provide_credentials` returns. In this case, `provide_credentials` returns no credentials.
```